pkg/deploy/chart_manager.go

```diff
@@ -74,81 +76,97 @@ type SkrChartConfig struct {
 
 func NewEnabledSKRWebhookChartManager(config *SkrChartConfig) *EnabledSKRWebhookChartManager {
 	return &EnabledSKRWebhookChartManager{
-		cache:  NewSKRChartClientCache(),
+		cache:  modulelib.NewRendererCache(),
 		config: config,
 	}
 }
 
-func (m *EnabledSKRWebhookChartManager) InstallWebhookChart(ctx context.Context, kyma *v1alpha1.Kyma,
-	remoteClientCache *remote.ClientCache, kcpClient client.Client,
+func (m *EnabledSKRWebhookChartManager) InstallWebhookChart(ctx context.Context,
+	syncContext *remote.KymaSynchronizationContext,
 ) (bool, error) {
-	skrClient, err := remote.NewRemoteClient(ctx, kcpClient, client.ObjectKeyFromObject(kyma),
-		kyma.Spec.Sync.Strategy, remoteClientCache)
+	logger := logf.FromContext(ctx)
+	kymaObjKey := client.ObjectKeyFromObject(syncContext.ControlPlaneKyma)
+	skrCfg, err := remote.GetRemoteRestConfig(ctx, syncContext.ControlPlaneClient, kymaObjKey,
```

Comment:
can we store the `rest.Config` also so the cache along with `client.Client`, since they belong together and we do not have to do a lookup for the secret everytime